authentication/authentication.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+import "common/common.proto";
+
+// Package authentication handles authentication such
+// as login, logout, and access token refresh.
+//
+// This package doesn't necessarily implements OAuth2
+// specification, but at least it borrows some of the
+// thing that defined by the OAuth2 specification.
+package authentication;
+
+option go_package = "tokobapak/protobuf/authenticationpb";
+option java_package = "id.my.tokobapak.protobuf";
+option java_outer_classname = "Auth";
+option java_multiple_files = true;
+option csharp_namespace = "TokoBapak.Protobuf.AuthenticationSchema";
+option cc_enable_arenas = true;
+
+// Authentication half-heartly implements OAuth2 specification.
+service Authentication {
+  // The usual login mechanism, it doesn't support 2FA challenge yet
+  // for the response (it can be done using oneof).
+  // Right now, we're focusing on the fastest way to develop.
+  rpc Login(LoginRequest) returns (TokenSetReply) {}
+  // Refresh acquires new TokenSet for expired access tokens.
+  rpc Refresh(RefreshRequest) returns (TokenSetReply) {}
+  // Sends a logout state for the user. The access and refresh token
+  // will be invalidated.
+  rpc Logout(LogoutRequest) returns (common.EmptyReply) {}
+}
```

Comment:
Do we need to add Forgot Password here ?

Reply:
no, that's beyond the scope of authentication.

Reply:
Got it, then we may change this architecture accordingly

![image](https://user-images.githubusercontent.com/39396616/211434295-b3e31197-708c-48c2-b5e9-04382ababe46.png)

Reply:
yes, I actually wanted to change and create more detailed version of TokoBapak architecture, yet I don't have the time for that right now. I'm pretty much preoccupied by my day job.

Reply:
you've done a lot bro, appreciate